roles/__bigip_selfip/tasks/main.yaml

```diff
@@ -39,6 +39,44 @@
       that:
           - not result|changed
 
+- name: Create Self IP, specify a Route Domain
+  bigip_selfip:
+      address: "{{ selfip }}"
+      name: "{{ selfip_name }}"
+      netmask: "{{ selfip_netmask }}"
+      password: "{{ bigip_password }}"
+      server: "{{ inventory_hostname }}"
+      server_port: "{{ bigip_port }}"
+      user: "{{ bigip_username }}"
+      validate_certs: "{{ validate_certs }}"
+      vlan: "{{ selfip_vlan }}"
+      route_domain: "{{ selfip_rd }}"
+  register: result
+
+- name: Assert Create Self IP, specify a Route Domain
+  assert:
+      that:
+          - result|changed
+
+- name: Create Self IP, specify a Route Domain - Idempotent check
+  bigip_selfip:
+      address: "{{ selfip }}"
+      name: "{{ selfip_name }}"
+      netmask: "{{ selfip_netmask }}"
+      password: "{{ bigip_password }}"
+      server: "{{ inventory_hostname }}"
+      server_port: "{{ bigip_port }}"
+      user: "{{ bigip_username }}"
+      validate_certs: "{{ validate_certs }}"
+      vlan: "{{ selfip_vlan }}"
+      route_domain: "{{ selfip_rd }}"
+  register: result
+
+- name: Assert Create Self IP, specify a Route Domain - Idempotent check
+  assert:
+      that:
+          - result|changed
```

Comment:
This should be "not result|changed" because you are checking for an idempotent operation; ie, you want to make sure that the exact same task does _not_ change the value if it has already set it.